base/strings/faces.jl

```diff
@@ -0,0 +1,698 @@
+# TOML stuff
+
+# """
+#     loadfaces!(facetoml::IO)
+
+# Parse `facetoml` as TOML, and load all faces described.
+# The loading is done with `loadfaces!`, which see.
+
+# Face entries should be of the following form:
+# ```toml
+# [face_name]
+# property = "value"
+# """
+# loadfaces!(facetoml::IO) = loadfaces!(TOML.parse(facetoml))
+
+# """
+#     loadfaces!(facetoml::String)
+
+# Open `facetoml`, parse it as TOML, and load all faces described.
+# The loading is done with `loadfaces!`, which see.
+# """
+# loadfaces!(facetoml::String) =
+#     open(facetoml, "r") do io loadfaces!(io) end
+
+const RGBTuple = NamedTuple{(:r, :g, :b), NTuple{3, UInt8}}
+
+"""
+    struct SimpleColor
+
+A basic representation of a color, intended for string styling purposes.
+It can either contain a named color (like `:red`), or an `RGBTuple` which
+is a NamedTuple specifying an `r`, `g`, `b` color with a bit-depth of 8.
+
+## Constructors
+
+```julia
+SimpleColor(name::Symbol)  # e.g. :red
+SimpleColor(rgb::RGBTuple) # e.g. (r=1, b=2, g=3)
+SimpleColor(r::Integer, b::Integer, b::Integer)
+SimpleColor(rgb::UInt32)   # e.g. 0x123456
+```
+
+Also see `tryparse(SimpleColor, rgb::String)`.
+"""
+struct SimpleColor
+    value::Union{Symbol, RGBTuple}
+end
+
+SimpleColor(r::Integer, g::Integer, b::Integer) = SimpleColor((; r=UInt8(r), g=UInt8(g), b=UInt8(b)))
+SimpleColor(rgb::UInt32) = SimpleColor(reverse(reinterpret(UInt8, [rgb]))[2:end]...)
+
+"""
+    tryparse(::Type{SimpleColor}, rgb::String)
+
+Attempt to parse `rgb` as a `SimpleColor`. If `rgb` starts with
+`#` and has a length of 7, it is converted into a `RGBTuple`-backed `SimpleColor`.
+If `rgb` starts with `a`-`z`, `rgb` is interpreted as a color name
+and converted to a `Symbol`-backed `SimpleColor`.
+
+Otherwise, `nothing` is returned.
+"""
+function tryparse(::Type{SimpleColor}, rgb::String)
+    if startswith(rgb, '#') && length(rgb) == 7
+        SimpleColor(parse(Int, rgb[2:3], base=16),
```

Comment:
I suppose I should be a bit stricter since we only want `[0-9a-f]`, say
```julia
ncodeunits(rgb) == 7 && first(rgb) == '#' && all(∈(('#',) ∪ ('0':'9') ∪ ('a':'f')), rgb)
```